relational_operators/AggregationOperator.hpp

```diff
@@ -103,23 +109,23 @@ class AggregationWorkOrder : public WorkOrder {
    * @brief Constructor
    *
    * @param input_block_id The block id.
-   * @param aggr_state_index The index of the AggregationState in QueryContext.
+   * @param state The AggregationState to use.
    **/
   AggregationWorkOrder(
       const block_id input_block_id,
-      const QueryContext::aggregation_state_id aggr_state_index)
+      AggregationOperationState *state)
```

Comment:
The main benefit is to remove the `Worker`'s dependencies on `QueryContext`, `CatalogDatabase`, and `StorageManager` to execute `WorkOrder`s.

The distributed case needs `WO` serializations, which could directly construct self-contained `WO`s, instead of the old version that needs `QueryContext`, `CatalogDatabase`, and `StorageManager` for their execution.

See [WorkOrderFactory](https://github.com/pivotalsoftware/quickstep/pull/36/files#diff-06aa2846a4548109684c9f6d61c90f0b) for more info.